feat(docs-infra): add docs-image extension for enhanced image handlin…

SkyZeroZx opened the pull request:
…g in markdown

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
 https://pagespeed.web.dev/analysis/https-angular-dev-events-v21/ynupjvkrx9?form_factor=mobile
<img width="976" height="292" alt="image" src="https://github.com/user-attachments/assets/1be4b0b4-a22b-47d8-a532-df5a650eb312" />



## What is the new behavior?
The loading of image resources should be improved when necessary.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

josephperrott reviewed (approved):
LGTM

JeanMeche commented:
Honnestly I'm not sure that's better than using a regular HTML `img` element.

SkyZeroZx commented:
> Honnestly I'm not sure that's better than using a regular HTML `img` element.

It depends on the case. Considering that when we use Markdown, the `img` tag is wrapped in a `p` tag and a custom class is added, it could be moderately useful for cases where we need to defer or quickly load an image.

JeanMeche commented:
Fair enough, I don't feel too strongly about it. 
Also, we'll need a rebase to fix the conflict.

JeanMeche commented:
We have a linting issue. Can you PTAL.

SkyZeroZx committed:
feat(docs-infra): add docs-image extension for enhanced image handling in markdown

Extends the Markdown parser to support `loading`, `decoding`, and `fetchpriority` attributes for images via curly brace syntax.
